File: backend/src/services/searchservice.js
import { embeddings } from "../config/embeddings.js";
import { qdrant } from "../config/qdrant.js";

export async function searchChunks({ documentId, userId, query }) {
  const vector = await embeddings.embedQuery(query);

  const result = await qdrant.search(process.env.QDRANT_COLLECTION, {
    vector,
    limit: 5,
    filter: {
      must: [
        { key: "documentId", match: { value: documentId } },
        { key: "userId", match: { value: userId } }
      ]
    }
  });

  return result;
}

File: backend/src/services/search_service.py
from langchain_core.documents import Document
from qdrant_client.models import FieldCondition, Filter, MatchValue

from src.integrations.vectorstore import get_vectorstore


def search_chunks(
    *,
    document_id: str,
    user_id: str,
    query: str,
    limit: int = 5,
) -> list[Document]:
    vectorstore = get_vectorstore()

    qdrant_filter = Filter(
        must=[
            FieldCondition(
                key="metadata.documentId",
                match=MatchValue(value=document_id),
            ),
            FieldCondition(
                key="metadata.userId",
                match=MatchValue(value=user_id),
            ),
        ]
    )

    return vectorstore.similarity_search(
        query=query,
        k=limit,
        filter=qdrant_filter,
    )